src/modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardEventHandlers.cpp

```diff
@@ -822,6 +930,708 @@ namespace KeyboardEventHandlers
+                toast(title, message);
+                return;
+            }
+
+            std::wstring executable_and_args = fmt::format(L"\"{}\" {}", fullExpandedFilePath, shortcut.runProgramArgs);
+
+            WCHAR currentDir[MAX_PATH];
+            WCHAR* currentDirPtr = currentDir;
+            DWORD result = ExpandEnvironmentStrings(shortcut.runProgramStartInDir.c_str(), currentDir, MAX_PATH);
+
+            if (shortcut.runProgramStartInDir == L"")
+            {
+                currentDirPtr = nullptr;
+            }
+            else
+            {
+                DWORD dwAttrib = GetFileAttributesW(currentDir);
+
+                if (dwAttrib == INVALID_FILE_ATTRIBUTES)
+                {
+                    std::wstring title = fmt::format(L"Error starting {}", fileNamePart);
+                    std::wstring message = fmt::format(L"The start in path was not valid. It could not be used.", currentDir);
+                    currentDirPtr = nullptr;
+                    toast(title, message);
+                    return;
+                }
+            }
+
+            DWORD processId = 0;
+            HANDLE newProcessHandle;
+
+            if (shortcut.elevationLevel == Shortcut::ElevationLevel::Elevated)
+            {
+                newProcessHandle = run_elevated(fullExpandedFilePath, shortcut.runProgramArgs, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+                processId = GetProcessId(newProcessHandle);
+            }
+            else if (shortcut.elevationLevel == Shortcut::ElevationLevel::NonElevated)
+            {
+                run_non_elevated(fullExpandedFilePath, shortcut.runProgramArgs, &processId, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+            }
+            else if (shortcut.elevationLevel == Shortcut::ElevationLevel::DifferentUser)
+            {
+                newProcessHandle = run_as_different_user(fullExpandedFilePath, shortcut.runProgramArgs, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+                processId = GetProcessId(newProcessHandle);
+            }
+
+            if (processId == 0)
+            {
+                std::wstring title = fmt::format(L"Error starting {}", fileNamePart);
+                std::wstring message = fmt::format(L"The application might not have started.");
+                toast(title, message);
+                return;
+            }
+
+            if (shortcut.startWindowType == Shortcut::StartWindowType::Hidden)
+            {
+                HideProgram(processId, fileNamePart, 0);
+            }
+            //ShowProgram(processId, fileNamePart, true, false, (shortcut.startWindowType == Shortcut::StartWindowType::Hidden), 0);
+        }
+        return;
+    }
+
+    void CloseAndTerminateProcessByName(const std::wstring& fileNamePart)
+    {
+        CloseProcessByName(fileNamePart);
+        Sleep(1000);
```

Comment:
Why the sleep here? This method doesn't even seem to be called.

Reply:
This is discussed here, https://github.com/microsoft/PowerToys/pull/30121#issuecomment-1868537818
in bullet 2.

I reinstated the call to the method for here since I think there is value. However, I also am fine with removing this from the select list and then also removing the code.

I am fine with removing it because if you want the "nice close" and then "terminate" you could just use two different shortcuts.

Reply:
I think I'm OK with it if it's another thread 😉

Reply:
Sounds great too!

Reply:
Thank you!